modules/finance/src/main/java/com/opengamma/strata/finance/rate/future/DeliverableSwapFuture.java

```diff
@@ -0,0 +1,585 @@
+/**
+ * Copyright (C) 2015 - present by OpenGamma Inc. and the OpenGamma group of companies
+ *
+ * Please see distribution for license.
+ */
+package com.opengamma.strata.finance.rate.future;
+
+import java.io.Serializable;
+import java.time.LocalDate;
+import java.util.Map;
+import java.util.NoSuchElementException;
+import java.util.Set;
+
+import org.joda.beans.Bean;
+import org.joda.beans.BeanDefinition;
+import org.joda.beans.ImmutableBean;
+import org.joda.beans.ImmutableValidator;
+import org.joda.beans.JodaBeanUtils;
+import org.joda.beans.MetaProperty;
+import org.joda.beans.Property;
+import org.joda.beans.PropertyDefinition;
+import org.joda.beans.impl.direct.DirectFieldsBeanBuilder;
+import org.joda.beans.impl.direct.DirectMetaBean;
+import org.joda.beans.impl.direct.DirectMetaProperty;
+import org.joda.beans.impl.direct.DirectMetaPropertyMap;
+
+import com.opengamma.strata.basics.currency.Currency;
+import com.opengamma.strata.collect.ArgChecker;
+import com.opengamma.strata.finance.Product;
+import com.opengamma.strata.finance.Security;
+import com.opengamma.strata.finance.rate.swap.ExpandedSwapLeg;
+import com.opengamma.strata.finance.rate.swap.NotionalExchange;
+import com.opengamma.strata.finance.rate.swap.NotionalPaymentPeriod;
+import com.opengamma.strata.finance.rate.swap.PaymentEvent;
+import com.opengamma.strata.finance.rate.swap.PaymentPeriod;
+import com.opengamma.strata.finance.rate.swap.Swap;
+import com.opengamma.strata.finance.rate.swap.SwapLeg;
+import com.opengamma.strata.finance.rate.swap.SwapLegType;
+
+/**
+ * A deliverable swap futures contract.
+ * <p>
+ * A deliverable swap future is a financial instrument that physically settles an interest rate swap on a future date. 
+ * The futures product is margined on a daily basis. 
```

Comment:
A deliverable swap future is a financial instrument that physically settles an interest rate swap on a future date. The delivered swap is cleared by a central counterparty. The last future price before delivery is quoted in term of the underlying swap present value. The futures product is margined on a daily basis.